### PR TITLE
Fix warning when record without type information

### PR DESCRIPTION
**Pull request by xinz**
Hello @dumbbell, 

Closes https://github.com/yakaz/yamerl/issues/38, could you please help on this when available?

Thanks!

### CODE REVIEW

**Comment by xinz** [comment]
The following CI build failed is not involved by commit https://github.com/yakaz/yamerl/pull/42/commits/3e5428dc5283c38ea537e6b6e30e7cfdfc18bcab , and I can reproduced it in my local, here is my local same error

```
➜  yamerl git:(fix-warning) ✗ rm -rf _build
➜  yamerl git:(fix-warning) ✗ ./rebar3 eunit
===> Verifying dependencies...
===> Analyzing applications...
===> Compiling yamerl
===> Performing EUnit tests...
===> Error Running EUnit Tests:
  Module `alias' not found in project.
  Module `bool' not found in project.
  Module `bool_json' not found in project.
  Module `document_delimiters' not found in project.
  Module `document_marker_only' not found in project.
  Module `empty' not found in project.
  Module `erlang_atom' not found in project.
  Module `erlang_atom_if_exist' not found in project.
  Module `erlang_atom_if_exist_auto' not found in project.
  Module `erlang_binary' not found in project.
  Module `erlang_fun' not found in project.
  Module `float' not found in project.
  Module `float_yaml11' not found in project.
  Module `ipaddr' not found in project.
  Module `map' not found in project.
  Module `null' not found in project.
  Module `null_json' not found in project.
  Module `seq' not found in project.
  Module `size' not found in project.
  Module `str' not found in project.
  Module `stream' not found in project.
  Module `timestamp' not found in project.
  Module `unicode_string' not found in project.
  Module `unknown_alias' not found in project.
  Module `unrecognized_tag' not found in project.
  Module `invoice' not found in project.
  Module `logfile' not found in project.
  Module `block_mapping_noeol' not found in project.
  Module `bom_between_docs' not found in project.
  Module `directives_end_alone' not found in project.
  Module `document_end_alone' not found in project.
  Module `dos_newlines' not found in project.
  Module `empty' not found in project.
  Module `enc_invalid_unicode' not found in project.
  Module `enc_utf16be_bom' not found in project.
  Module `enc_utf16be_char' not found in project.
  Module `enc_utf16le_bom' not found in project.
  Module `enc_utf16le_char' not found in project.
  Module `enc_utf32be_bom' not found in project.
  Module `enc_utf32be_char' not found in project.
  Module `enc_utf32le_bom' not found in project.
  Module `enc_utf32le_char' not found in project.
  Module `enc_utf8_bom' not found in project.
  Module `enc_utf8_char' not found in project.
  Module `ex_5.10_reserved_indicators' not found in project.
  Module `ex_5.12_tabs_and_spaces' not found in project.
  Module `ex_5.13_escaped_characters' not found in project.
  Module `ex_5.14_invalid_escaped_characters' not found in project.
  Module `ex_5.2_invalid_bom' not found in project.
  Module `ex_5.3_block_structure_indicators' not found in project.
  Module `ex_5.4_flow_collection_indicators' not found in project.
  Module `ex_5.6_node_property_indicators' not found in project.
  Module `ex_5.7_block_scalar_indicators' not found in project.
  Module `ex_5.8_quoted_scalar_indicators' not found in project.
  Module `ex_5.9_directive_indicator' not found in project.
  Module `ex_6.11_multiline_comments' not found in project.
  Module `ex_6.12_separation_spaces' not found in project.
  Module `ex_6.13_reserved_directives' not found in project.
  Module `ex_6.14_yaml_directive' not found in project.
  Module `ex_6.15_invalid_repeated_yaml_directive' not found in project.
  Module `ex_6.16_tag_directive' not found in project.
  Module `ex_6.17_invalid_repeated_tag_directive' not found in project.
  Module `ex_6.18_primary_tag_handle' not found in project.
  Module `ex_6.19_secondary_tag_handle' not found in project.
  Module `ex_6.1_indentation_spaces' not found in project.
  Module `ex_6.20_tag_handles' not found in project.
  Module `ex_6.21_local_tag_prefixes' not found in project.
  Module `ex_6.22_global_tag_prefix' not found in project.
  Module `ex_6.23_node_properties' not found in project.
  Module `ex_6.24_verbatim_tags' not found in project.
  Module `ex_6.25_invalid_verbatim_tags' not found in project.
  Module `ex_6.26_tag_shorthands' not found in project.
  Module `ex_6.27_invalid_tag_shorthands' not found in project.
  Module `ex_6.28_non_specific_tags' not found in project.
  Module `ex_6.29_node_anchors' not found in project.
  Module `ex_6.2_indentation_indicators' not found in project.
  Module `ex_6.3_separation_spaces' not found in project.
  Module `ex_6.4_line_prefixes' not found in project.
  Module `ex_6.5_empty_lines' not found in project.
  Module `ex_6.6_line_folding' not found in project.
  Module `ex_6.7_block_folding' not found in project.
  Module `ex_6.8_flow_folding' not found in project.
  Module `ex_6.9_separated_comment' not found in project.
  Module `ex_7.10_plain_characters' not found in project.
  Module `ex_7.11_plain_implicit_keys' not found in project.
  Module `ex_7.12_plain_lines' not found in project.
  Module `ex_7.13_flow_sequence' not found in project.
  Module `ex_7.14_flow_sequence_entries' not found in project.
  Module `ex_7.15_flow_mapping' not found in project.
  Module `ex_7.16_flow_mapping_entries' not found in project.
  Module `ex_7.17_flow_mapping_separate_values' not found in project.
  Module `ex_7.18_flow_mapping_adjacent_values' not found in project.
  Module `ex_7.19_single_pair_flow_mappings' not found in project.
  Module `ex_7.20_single_pair_explicit_entry' not found in project.
  Module `ex_7.21_single_pair_implicit_entries' not found in project.
  Module `ex_7.22_invalid_implicit_keys' not found in project.
  Module `ex_7.23_flow_content' not found in project.
  Module `ex_7.24_flow_nodes' not found in project.
  Module `ex_7.2_empty_content' not found in project.
  Module `ex_7.3_completely_empty_flow_nodes' not found in project.
  Module `ex_7.4_double_quoted_implicit_keys' not found in project.
  Module `ex_7.5_double_quoted_line_breaks' not found in project.
  Module `ex_7.6_double_quoted_lines' not found in project.
  Module `ex_7.7_single_quoted_characters' not found in project.
  Module `ex_7.8_single_quoted_implicit_keys' not found in project.
  Module `ex_7.9_single_quoted_lines' not found in project.
  Module `ex_8.10_folded_lines' not found in project.
  Module `ex_8.14_block_sequence' not found in project.
  Module `ex_8.15_block_sequence_entry_types' not found in project.
  Module `ex_8.16_block_mappings' not found in project.
  Module `ex_8.17_explicit_block_mapping_entries' not found in project.
  Module `ex_8.18_implicit_block_mapping_entries' not found in project.
  Module `ex_8.19_compact_block_mappings' not found in project.
  Module `ex_8.1_block_scalar_header' not found in project.
  Module `ex_8.20_block_node_types' not found in project.
  Module `ex_8.21_block_scalar_nodes' not found in project.
  Module `ex_8.22_block_collection_nodes' not found in project.
  Module `ex_8.2_block_indentation_indicator' not found in project.
  Module `ex_8.3_invalid_block_scalar_indentation_indicators_1' not found in project.
  Module `ex_8.3_invalid_block_scalar_indentation_indicators_2' not found in project.
  Module `ex_8.3_invalid_block_scalar_indentation_indicators_3' not found in project.
  Module `ex_8.4_chomping_final_line_break' not found in project.
  Module `ex_8.5_chomping_trailing_lines' not found in project.
  Module `ex_8.6_empty_scalar_chomping' not found in project.
  Module `ex_8.7_literal_scalar' not found in project.
  Module `ex_8.8_literal_content' not found in project.
  Module `ex_8.9_folded_scalar' not found in project.
  Module `ex_9.1_document_prefix' not found in project.
  Module `ex_9.2_document_markers' not found in project.
  Module `ex_9.3_bare_documents' not found in project.
  Module `ex_9.4_explicit_documents' not found in project.
  Module `ex_9.5_directives_documents' not found in project.
  Module `ex_9.6_stream' not found in project.
  Module `flow_mapping_empty_1' not found in project.
  Module `flow_mapping_empty_10' not found in project.
  Module `flow_mapping_empty_11' not found in project.
  Module `flow_mapping_empty_12' not found in project.
  Module `flow_mapping_empty_2' not found in project.
  Module `flow_mapping_empty_3' not found in project.
  Module `flow_mapping_empty_4' not found in project.
  Module `flow_mapping_empty_5' not found in project.
  Module `flow_mapping_empty_6' not found in project.
  Module `flow_mapping_empty_7' not found in project.
  Module `flow_mapping_empty_8' not found in project.
  Module `flow_mapping_empty_9' not found in project.
  Module `flow_mapping_no_entry_1' not found in project.
  Module `maps_and_custom_tags' not found in project.
  Module `no_such_file' not found in project.
  Module `noeol' not found in project.
  Module `non_ascii_line_breaks_yaml11' not found in project.
  Module `non_ascii_line_breaks_yaml12' not found in project.
  Module `option_doc_version_forced' not found in project.
  Module `option_doc_version_invalid' not found in project.
  Module `option_doc_version_ok' not found in project.
  Module `option_doc_version_too_high' not found in project.
  Module `option_doc_version_too_low' not found in project.
  Module `option_io_blocksize_invalid' not found in project.
  Module `option_io_blocksize_ok' not found in project.
  Module `option_token_fun_invalid' not found in project.
  Module `option_token_fun_ok' not found in project.
  Module `string_binary' not found in project.
  Module `string_list' not found in project.
  Module `unsupported_yaml_version_13' not found in project.
  Module `unsupported_yaml_version_20' not found in project.
  Module `decode' not found in project.
  Module `decode_file' not found in project.
```

We can fix this issue to use `rebar3 eunit` instead of `./rebar3 eunit`, here are my local result with this commit https://github.com/yakaz/yamerl/pull/42/commits/a15ae0fae9b9bd194da260ae884085f9264495be, please let me know if any missing.

```
➜  yamerl git:(fix-warning) ✗ rm -rf _build
➜  yamerl git:(fix-warning) ✗ rebar3 eunit
===> Verifying dependencies...
===> Compiling yamerl
===> Performing EUnit tests...
.........................................................................................................................................................................................................................................................
Finished in 2.178 seconds
249 tests, 0 failures
```

```
➜  yamerl git:(fix-warning) ✗ rm -rf _build
➜  yamerl git:(fix-warning) ✗ rebar3 do compile,eunit,dialyzer
===> Verifying dependencies...
===> Compiling yamerl
===> Verifying dependencies...
===> Compiling yamerl
===> Performing EUnit tests...
.........................................................................................................................................................................................................................................................
Finished in 2.233 seconds
249 tests, 0 failures
===> Verifying dependencies...
===> Compiling yamerl
===> Dialyzer starting, this may take a while...
===> Updating plt...
===> Resolving files...
===> Updating base plt...
===> Resolving files...
===> Checking 197 files in "/Users/xinz/.cache/rebar3/rebar3_22.3.4.2_plt"...
===> Copying "/Users/xinz/.cache/rebar3/rebar3_22.3.4.2_plt" to "/Users/xinz/workspace/github/xinz/yamerl/_build/default/rebar3_22.3.4.2_plt"...
===> Checking 197 files in "/Users/xinz/workspace/github/xinz/yamerl/_build/default/rebar3_22.3.4.2_plt"...
===> Doing success typing analysis...
===> Resolving files...
===> Analyzing 28 files with "/Users/xinz/workspace/github/xinz/yamerl/_build/default/rebar3_22.3.4.2_plt"...
```

File: include/internal/yamerl_constr.hrl [hunk 1]
```diff
@@ -43,14 +43,14 @@
...
 
 -record(unfinished_node, {
-    module = ?MODULE,
-    path,
-    pres,
-    priv
+    module = ?MODULE :: atom(),
+    path  :: tuple(),
+    pres  :: [tuple()],
+    priv  :: tuple() | map() | list()
```
Comment (dumbbell) [inline include/internal/yamerl_constr.hrl:49]: Perhaps this should be `any()` instead of `tuple() | map() | list()`, what do you think?

**Review by dumbbell — COMMENTED** [review]
I tested it successfully locally, thank you. I just have one comment and would like your opinion :-)

File: include/internal/yamerl_constr.hrl (hunk 1 above)
Comment (xinz) [reply include/internal/yamerl_constr.hrl]: Thank you, agree, I have committed this change, please review again.

Btw, if we can release a new version that would be appreciated.

**Comment by dumbbell** [comment]
> Btw, when all is ready, if we can release a new version, that's much appreciated.

@xinz: Version 0.8.1 was just published. Thank you!

**Pull request merged**